pkgs/development/compilers/gcc/11/default.nix

```diff
@@ -70,10 +64,13 @@ let majorVersion = "11";
       ++ optional langFortran ../gfortran-driving.patch
       ++ optional (targetPlatform.libc == "musl" && targetPlatform.isPower) ../ppc-musl.patch
 
-      ++ optional (stdenv.isDarwin && stdenv.isAarch64) (fetchpatch {
-        url = "https://github.com/fxcoudert/gcc/compare/releases/gcc-11.2.0...gcc-11.2.0-arm-20211201.diff";
-        sha256 = "sha256-z62s/cXuH9Kgq/oD/OiiZ8LWnX1xl1D43sONnwaEW1w=";
-      })
+      ++ optionals stdenv.isDarwin [
+        (fetchpatch {
+          # Corresponds to https://github.com/iains/gcc-11-branch/releases/tag/gcc-11.3-darwin-r2
+          url = "https://github.com/Homebrew/formula-patches/raw/8184603db1db7da12d0fbfa47e1329ba0e21558e/gcc/gcc-11.3.0-arm.diff";
```

Comment:
It's a bit unclear how you go from the URL in the comment to the one in the code.

Previously it was a bit more obvious with only tags in the URL and the actual source repo rather than homebrew derivative patch.

Reply:
It's from the Homebrew maintainer's [commit message](https://github.com/Homebrew/formula-patches/commit/07e71538ad8adbf6cf570ebcd03851eda02cf10a). Since [our gcc12](https://github.com/NixOS/nixpkgs/pull/189754) uses Homebrew's prepared patch I thought it's simpler to go with theirs as well.

Reply:
My point is that there are implicit steps that you need to take to get the latest URL.

You link to https://github.com/iains/gcc-11-branch tag page. There is no connection to homebrew URL from there. How will the next person know where to get the patch?

GCC 12 used to point to fxcoudert's repo diff too, until it was changed in 74b420b612c, 657075f9fa599, and 816cb20bd3. I don't think it was the right decision. Homebrew isn't the source of these patches and it doesn't add anything on top, so you might as well use iains's repos where it's obvious how to get the diff from just looking at the URL. You also get the exact tag right in the URL, whereas for homebrew you need extra steps to learn it.

Reply:
Ok, what do you think of the following URL?

> 
https://github.com/iains/gcc-11-branch/compare/2d280e7eafc086e9df85f50ed1a6526d6a3a204d..gcc-11.3-darwin-r2.diff

(`2d280e7` is `releases/gcc-11.3.0` but we can't refer to it by name since it isn't tagged in the repo and it's not a fork)

Reply:
With a comment like this it would be great:

```nix
# There are no upstream release tags in https://github.com/iains/gcc-11-branch.
# 2d280e7 is the commit from https://github.com/gcc-mirror/gcc/releases/tag/releases%2Fgcc-11.3.0
```